server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServerResourceManager.java

```diff
@@ -556,14 +557,15 @@ private void manageMemory() {
                 continue;
               }
               Tablet tablet = tabletReport.getTablet();
-              if (!tablet.initiateMinorCompaction(MinorCompactionReason.SYSTEM)) {
-                if (tablet.isClosed()) {
+              var state = context.getTableManager().getTableState(tablet.getExtent().tableId());
+              if (!tablet.initiateMinorCompaction(MinorCompactionReason.SYSTEM, state)) {
+                if (tablet.isClosed() || state.equals(TableState.DELETING)) {
```

Comment:
Enums should always be compared using `==` instead of `.equals()`:

```suggestion
                if (tablet.isClosed() || state == TableState.DELETING) {
```